docker/base.dockerfile

```diff
@@ -1,51 +1,46 @@
 FROM ubuntu:16.04
-RUN apt-get -qq update
-
-RUN apt-get install -qq -y software-properties-common python-software-properties
 
+# Install dependencies
+RUN apt-get update -qq
+RUN apt-get install -qq -y software-properties-common
 RUN add-apt-repository -y ppa:deadsnakes/ppa
 RUN apt-get update -qq
-
 RUN apt-get -qq install -y \
-                build-essential \
-                python3.6 \
-                python3.6-dev \
-                python3-pip \
-                python3.6-venv \
-                virtualenv \
-                git
-
-# update pip
-RUN python3.6 -m pip install pip --upgrade
-RUN python3.6 -m pip install wheel
-
-RUN apt install -qq -y \
-            git \
-            libglib2.0-0 \
-            libsm6 \
-            libxrender1 \
-            default-jre
-
-
+  build-essential \
+  default-jre \
+  git \
+  libglib2.0-0 \
+  libsm6 \
+  libxrender1 \
+  python-pip \
+  python-software-properties \
+  python2.7 \
+  python2.7-dev \
+  python3-pip \
+  python3.6 \
+  python3.6-dev \
+  python3.6-venv \
+  virtualenv \
+  && rm -rf /var/lib/apt/lists/*
+
+# Temporarily add setup.py, needed for env setup.
 RUN mkdir /petastorm
-
-ADD README.rst /petastorm/
 ADD setup.py /petastorm/
+ADD README.rst /petastorm/
 ADD petastorm /petastorm/petastorm
 
+# Set up Python3 environment
+RUN python3.6 -m pip install pip --upgrade
+RUN python3.6 -m pip install wheel
 RUN python3.6 -m venv /petastorm_venv3.6
 RUN /petastorm_venv3.6/bin/pip3.6 install -e /petastorm/[test,tf,torch,docs,opencv]
 RUN /petastorm_venv3.6/bin/pip3.6 uninstall -y petastorm
```

Comment:
As far as I can tell, this line (and the repeated one in the Python3.6 section) don't actually do anything.